packages/baggage-span-processor/README.md

```diff
@@ -0,0 +1,63 @@
+# OpenTelemetry Baggage Span Processor for Node.js
+
+[![NPM Published Version][npm-img]][npm-url]
+[![Apache License][license-image]][license-url]
+
+The BaggageSpanProcessor reads entries stored in Baggage from the parent context and adds the [baggage](https://opentelemetry.io/docs/concepts/signals/baggage) entries' keys and
+values to the span as attributes on span start.
+
+Add this span processor to a tracer provider.
+
+Keys and values added to Baggage will appear on subsequent child spans for a trace within this service *and* be propagated to external services in accordance with any configured propagation formats configured.
+If the external services also have a Baggage span processor, the keys and values will appear in those child spans as well.
+
+⚠ Warning ⚠️
+
+Do not put sensitive information in Baggage.
+
+To repeat: a consequence of adding data to Baggage is that the keys and values will appear in all outgoing HTTP headers from the application.
```

Comment:
We have considered similar approaches to try to limit the baggage entries that are capture. In the equivalent [.NET](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1659) & [Java](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1290) PRs, we discussed allows and deny lists and/or prefixes, which felt a little cumborsome.

I like the idea of allowing the user to provide a function to determine if a given key should be added. It will then be customisable for the user instead the provider. I think this should be added as an enhancement so created the following issue to track. It would be preferable for that to be consistent across implementations if possible.
- https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2166

Reply:
Sure, I understand and thanks for your feedback. Once I've collected feedback on all implementations in each SDK contrib project, I plan on updating each to offer the same behaviour to control what keys are copied.

For reference, the above issue is linked to corresponding issues to discuss the user-defined key predicate function you suggested.